### ISSUE TITLE
Climatic Inventory plot

### ISSUE THREAD

**Issue by rdstern** [africanmathsinitiative/R-Instat#2724]
Here are some useful examples of the inventory graph that we had in the previous R version.  They are really useful and we don't yet have these options - nowhere near - with the ggplot2 version.

The first is rainfall for many stations.  This will usually be a graph from a single column.  The second is multiple elements.  they will usually be in separate columns.  It is also multiple stations, but that is a luxury.
The third is a different theme for the first one, and without labels on the right.

In the old format we didn't have facets - which is nice here.  I also quite like the addition of the rain/no-rain option, but that is a luxury.  This is mainly to see where data are missing.

I would very much like to see a full version of this facility by Version 0.3 on 20 March.  We could use it then!

<img width="792" alt="paper1" src="https://cloud.githubusercontent.com/assets/11226469/23329706/480e2f84-fb35-11e6-912f-3c0b4ab05eb8.png">
<img width="792" alt="paper2" src="https://cloud.githubusercontent.com/assets/11226469/23329709/535a0d2c-fb35-11e6-81ab-5c2b65652cd3.png">

![inventory for rainfall](https://cloud.githubusercontent.com/assets/11226469/23329761/686f9b90-fb36-11e6-8fe3-699250bbd05e.png)

**Comment by rdstern** [comment]
So what should the new dialogue look like.  I am happy to have it assume the following:
1) This isn't to be restrictive, as to be simple for the user.
2) The same element is in a single column, i.e. not multiple columns for say rainfall.
3) If there are multiple stations then the station is a factor in its own column.
4) Multiple elements are in different columns, e.g. rain, tmin, tmax.
5) A date column has been defined
6) The **Climatic > Define Climatic Data** has been used if needed.

The current R-Instat inventory plot has year as one axis and day-of-year as the other.  Thus it is only for a single element at a single station.  Hence it assumes that the date is a unique column.  Also effectively the date is on both axes.  Could we just give (automatically) the date column and let it calculate the tow columns it needs for the year and day-of-year?
 
The other plots, e.g. shown above have date on one axis and station, or element, or both on the other axis.

I assume we only want a single dialogue.  hence at the top we probably want (for now?) my famous

**Comment by maxwellfundi** [comment]
@rdstern  I feel the title, could be used from the plots sub dialog or is there a specific reason for this dialog that will need us to do it on the main dialog.

**Comment by rdstern** [comment]
I am really happy if we can fit as much as possible into our ordinary "system".  So yes, please proceed in whatever way you find most consistent to produce that command.

I assume we will also have to add that package.

**Comment by dannyparsons** [comment]
I think we will continue to do this as our own method since there is some internal manipulation of the data that is needed. This means like for Windroses we will produce a ggplot, but we won't have the flexibility to edit it through the initial dialog.

**Comment by maxwellfundi** [comment]
@rdstern  @stevekogo  and I are working on this and we would like more clarification on the Show Non-Missing checkbox. 
Its not clear to us what it is supposed to do. 

Thanks.

**Comment by rdstern** [comment]
I like the look of the new dialogue, which has now been merged.  I tried with the Dodoma data from the library, after making a date column.  But it doesn't recognise that column as acceptable in the dialogue.

**Comment by maxwellfundi** [comment]
The data needs to be defined as climatic before it's used in the inventory
plot dialog.

**Comment by rdstern** [comment]
Oooh, of course.  Now it works.  I like the results!  Now let's try some more!

**Comment by maxwellfundi** [comment]
Perfect, an quite happy you are impressed by that.

On Mar 18, 2017 5:05 PM, "rdstern" <notifications@github.com> wrote:

Oooh, of course. Now it works. I like the results! Now let's try some more!

—
You are receiving this because you were assigned.
Reply to this email directly, view it on GitHub
<https://github.com/africanmathsinitiative/Instat/issues/2724#issuecomment-287548447>,
or mute the thread

**Comment by rdstern** [comment]
OK, so I have tried this in a number of different ways and it seems to work fine so far.  Congratulations.  I tried with 2 sets, namely Dodoma with 4 elements and one station, followed by James data with about 40 stations and just rainfall.  I also tried with all 3 ways of doing graphs - using the Tools > Options menu.

Of course this is enough for now, but I would like to know your plans for the dialogue, partly because I suggest it will be good to continue discussions , especially with Steve, and also with the others while you are together.  So my list (not for Version 0.3!) includes the following:
First with the current type of display:
1) Flexibility in the legend, certainly of not having it, and certainly not having it called Recode.
2) Flexibility with the Y axis labels (DOY) and x axis, including not having the labels, e.g. "Year" is obvious!
3) Of course choice of colours.
Second with more options on the type of display.
1) I still like the option of a single, more complicated, plot, where there is a single row for each station and the date is just on one axis - so like the examples given at the top of this issue.  Do you have plans for this.  It makes it much easier to see, for example, whether max temp and min temp have the same missing sections or are different.
2) In the longer term I would also like the idea of a "label" on the left and right, where one might come from the linked data frame, e.g. latitude.  An example is given above.
3) I would also like to be able to sort the order of the stations/graphs, though this might be through the Prepare menu.

Cheers

**Comment by rdstern** [comment]
Inventory plot requested by Caribbean team in the form shown above, i.e. for data from multiple stations in a single plot.  Also with the information from the station data frame as labels, so that the name is on the left and the district could be on the right of the graph.

**Comment by stevekogo** [comment]
@rdstern Let me work on this then update you on how it goes.

**Comment by rdstern** [comment]
Great.  Thanks.  Looking forward.

**Comment by dannyparsons** [comment]
How about this?

![image](https://cloud.githubusercontent.com/assets/11214914/26014337/c8e010c6-3764-11e7-9a34-c16b87cd7ab3.png)
from this code:
```
kenya$Rain <- factor(!is.na(kenya$Rain), labels = c("Present", "Missing"), levels = c(TRUE, FALSE))
ggplot(kenya, aes(x = Date, color = Rain)) + 
  geom_linerange(aes(ymin = 0, ymax = 1)) + theme_bw() + 
  theme(axis.text.y = element_blank(), axis.ticks.y = element_blank(), axis.line.y = element_blank()) + 
  facet_wrap(~Station, ncol = 1) + 
  scale_colour_manual(values = c("Present" = "blue", "Missing" = "red")) + 
  scale_x_date(date_minor_breaks = "1 year") + 
  scale_y_continuous(breaks = NULL)
```
This will work for multiple stations or elements. Haven't though how to do both yet. @volloholic any ideas?

**Comment by dannyparsons** [comment]
I can do both stations and elements:

![image](https://cloud.githubusercontent.com/assets/11214914/26016005/f44ab1a6-376b-11e7-982f-3ba71609d62d.png)

using some stacking:
```
kenya <- reshape2::melt(kenya, measure.vars = c("Rain", "Tmax", "Tmin"))
kenya$Values <- factor(!is.na(kenya$value), labels = c("Present", "Missing"), levels = c(TRUE, FALSE))
ggplot(kenya, aes(x = Date, color = Values)) + 
  geom_linerange(aes(ymin = 0, ymax = 1)) + theme_bw() + 
  theme(axis.text.y = element_blank(), axis.ticks.y = element_blank(), axis.line.y = element_blank()) + 
  facet_grid(Station + variable~.) + 
  scale_colour_manual(values = c("Present" = "blue", "Missing" = "red")) + 
  scale_x_date(date_minor_breaks = "1 year") + 
  scale_y_continuous(breaks = NULL)
```

again this is facets tough. I'll see if I can find a way of using the y axis instead.

**Comment by dannyparsons** [comment]
I don't think will be possible with `geom_linerange` without using facets.

**Comment by dannyparsons** [comment]
Found an alternative with `Station` on y axis:

![image](https://cloud.githubusercontent.com/assets/11214914/26023063/20390582-37bd-11e7-9613-cd5b1154e98b.png)

Just used geom_raster instead:
```
ggplot(kenya, aes(x = Date, y = Station, fill = Rain)) + 
  geom_raster() + theme_bw() + 
  theme() + 
  scale_colour_manual(values = c("Present" = "blue", "Missing" = "red")) + 
  scale_x_date(date_minor_breaks = "1 year") + 
  geom_hline(yintercept = seq(0.5, by = 1, length.out = length(levels(kenya$Station)) + 1))
```

**Comment by dannyparsons** [comment]
Then for both elements and stations facet by one of them:
![image](https://cloud.githubusercontent.com/assets/11214914/26023127/2d7d27c2-37be-11e7-815a-7b35822325ea.png)

```
kenya <- reshape2::melt(kenya, measure.vars = c("Rain", "Tmax", "Tmin"), variable.name = "elements")
kenya$Key <- factor(!is.na(kenya$value), labels = c("Present", "Missing"), levels = c(TRUE, FALSE))
ggplot(kenya, aes(x = Date, y = elements, fill = Key)) + 
  geom_raster() + theme_bw() + 
  facet_grid(Station~.) + 
  scale_colour_manual(values = c("Present" = "blue", "Missing" = "red")) + 
  scale_x_date(date_minor_breaks = "1 year") +
  geom_hline(yintercept = seq(0.5, by = 1, length.out = length(levels(kenya$elements)) + 1))
```

**Comment by stevekogo** [comment]
@dannyparsons This looks nice and seems to be what Roger  has been keen to have.

**Comment by rdstern** [comment]
This looks great - yes just what we want.
There would also be some obvious options, which could be explored:
1. If just one out of station or element is given, then it doesn't need to have facets.
2. And often the one will be rainfall - so could use James data to explore.
3. With just one of station or element could alternatively have facets with the day of year on the y-axis - as we had before.
4. Or the day of year could sometimes be on the x-axis.  (Perhaps a coord-flip would then work.)
5. With just one station and only one element, then the original system would enable more detail to be seen.  (This could (perhaps) include the rain/no rain - or even more splits.)
6. I also like the way it currently distinguishes between data, missing and data not present (i.e. blank).  That helps us to see about the filling in.
7. There would be an "Inventory Options" button to go to a special sub-dialogue - as we have for boxplot, etc.  There would also be a way to to to the general graphics sub-dialogue if needed, i.e. this would be consistent with the other "Special" graphs.

**Comment by rdstern** [comment]
This is perhaps our first example of a "tailored" graph!
And this is enabling us to help users (mainly ggplot users) to understand "our" graphics system.  We have 3 types of graph, namely General, Special and Tailored.
**General** is to help users exploit the full power of ggplot.
**Special** is designed to make some common geoms easier to use - like boxplots or bar charts.
**Tailored** is designed to produce presentation graphs simply for specific purposes

Often **tailored** would be using a combination of layers (geoms) automatically.  In that sense I think that the sjplot **Describe > One Variable > Frequency**, etc could be classed as tailored and perhaps our simple graphs for **Describe > One Variable > Graph** also - just!

**Comment by dannyparsons** [comment]
fixed by #3070
